model/histogram/float_histogram.go

```diff
@@ -269,7 +269,7 @@ func (h *FloatHistogram) Add(other *FloatHistogram) *FloatHistogram {
 	h.Sum += other.Sum
 
 	if other.Schema != h.Schema {
-		other = other.ReduceResolution(h.Schema)
+		other = other.CopyToSchema(h.Schema)
```

Comment:
Alternative solution might be to pass `other` as value rather than as pointer:
`func (h *FloatHistogram) Add(other FloatHistogram) *FloatHistogram`